### ISSUE TITLE
False-positive in react/prop-types

### ISSUE THREAD

**Issue by wereHamster** [yannickcr/eslint-plugin-react#698]
```javascript
export function SomeComponent() {
  function f({foo}) {}

  return <div className={f()} />;
}
```

Somehow eslint thinks that the `{foo}` is a component prop:

> 'foo' is missing in props validation

It seems that the detection what actually is a prop is not precise enough. Because this warning/error can be avoided when I don't destruct the argument of the `f` function and instead access properties on it with the dot-notation:

```javascript
  function f(x) { x.foo ... }
```

Or if I destruct the argument inside the function body:

```javascript
  function f(x) { const {foo} = p; }
```

All three definitions of the `f` function are semantically equivalent, but this eslint plugin treats them differently.

**Comment by wereHamster** [comment]
Similar to #504
